app/src/main/java/edu/wpi/first/shuffleboard/app/components/WidgetPane.java

```diff
@@ -105,8 +107,10 @@ public WidgetPane() {
    *
    * @param predicate the predicate to use to find the desired widget tile
    */
-  public Optional<WidgetTile> tileMatching(Predicate<WidgetTile> predicate) {
+  public Optional<Tile> tileMatching(Predicate<Tile> predicate) {
     return tiles.stream()
+                .map(TypeUtils.optionalCast(Tile.class))
+                .filter(Optional::isPresent).map(Optional::get)
```

Comment:
May also want to add a `TypeUtils.optionalStream` method to flatmap the returned `Optional` as a stream like:

```java
.flatMap(TypeUtils::optionalStream)
```

Reply:
I agree! This seems to be a common pattern that this code base utilizes a lot. Turning it into a utility makes a lot of sense.